lib/worker.py

```diff
@@ -329,24 +332,32 @@ def run_loop(self, task_queue, result_queue, is_fragile):
                 break
 
             short_status = None
+            result_checksum = None
             result_queue.put(self.current_task(task_id))
-            if is_fragile:
+            testname = os.path.basename(task_id[0])
+            fragile_checksums = self.suite.get_test_fragile_checksums(testname)
+            if is_fragile and fragile_checksums:
```

Comment:
Here you block ability to rerun a test without a checksum. It is possible that a test cannot be verified just using a result file, but we want to allow reruns for it.

I remember the reason why you don't want to enable reruns by default when no checksums are provided: because it means that existing configs will let tests that are currently in a fragile list being retried.

However this is not so until we'll set `retries` explicitly, so it seems we can do this without subtle behaviour changes. I mean, if we'll consider empty (or non-existing) checksums list as 'rerun it anyway', everything will be good. But if you prefer explicit '*' in checksum, I don't mind.

You can also consider this as out of scope (not needed right now), but for me it looks quite natural that…

```json
fragile = {
    "retries": 2,
    "tests": {
        "bitset.test.lua": {
            "issues": ["gh-4095"]
        }
    }
}
```

…means 'rerun bitset.test.lua two times'.

Anyway, it is up to you. I just noted that the problem you highlighted looks as not applicable to our situation.

Reply:
I've commented it in the same time when you wrote the questions in here:
https://github.com/tarantool/test-run/pull/217#issuecomment-698128881
in short: I agree with it, but as standalone commit for this feature.